snappy/snappy.go

```diff
@@ -0,0 +1,69 @@
+package snappy
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"sync"
+
+	"github.com/golang/snappy"
+	"google.golang.org/grpc/encoding"
+)
+
+// Name is the name registered for the snappy compressor.
+const Name = "snappy"
+
+type compressor struct {
+	writerPool sync.Pool
+	readerPool sync.Pool
+}
+
+func init() {
+	c := &compressor{}
+	c.writerPool.New = func() interface{} {
+		return &writer{Writer: snappy.NewBufferedWriter(ioutil.Discard), pool: &c.writerPool}
+	}
+	c.readerPool.New = func() interface{} {
+		return &reader{Reader: snappy.NewReader(bytes.NewReader(nil)), pool: &c.readerPool}
+	}
+	encoding.RegisterCompressor(c)
```

Comment:
Oh interesting. I take it this registers the snappy implementation for the test servers as well, so we've already tested that this works!